R/calc_gdd_doy.R

```diff
@@ -22,49 +46,28 @@ calc_gdd_doy <- function(rast_dir, roi, gdd_threshold, gdd_base = 0) {
     files |> 
     fs::path_file() |>
     fs::path_ext_set(".bil")
-  rast_paths <- fs::path("/vsizip", files, bils)
-
-  #read in as raster stacks
-  prism <- terra::rast(rast_paths)
-  names(prism) <- doys
-  units(prism) <- "ºC"
+  rast_paths <- paste0("/vsizip/", fs::path(files, bils))
   
+  #read in multi-layer rasters
+  prism <- terra::rast(here::here(rast_paths))
+  names(prism) <- doys
+  terra::units(prism) <- "ºC"
   #sort layers by DOY
-  prism <- subset(prism, as.character(min(doys):max(doys)))
-  
-  #crop to roi
-  roi <- project(roi, prism)
-  prism_ne <- crop(prism, roi, mask = TRUE)
-  
-  # convert to degree days
-  # function for a single layer:
-  calc_dd <- function(tmean, base = 0) {
-    if (base != 0) {
-      dd <- tmean - base
-    } else {
-      dd <- tmean
-    }
-    dd[dd < 0] <- 0
-    dd
+  prism <- terra::subset(prism, as.character(min(doys):max(doys)))
+  #return
+  prism
+}
+
+#' Simple averaging method for GDD calculation
+#' 
+#' @param tmean Numeric vector; mean daily temp in ºC.
+#' @param base Base temp in ºC.
+calc_gdd_simple <- function(tmean, base = 0) {
+  if (base != 0) {
+    gdd <- tmean - base
+  } else {
+    gdd <- tmean
```

Comment:
I wonder if this is actually faster than just doing tmean - base regardless.  Overhead of if else might be greater than penalty for unecessarily subtracting 0